### PR TITLE
Fix stroke for rect markers

### LINKED ISSUE #2880: marker stroke broken for `marker=Rect`
```julia
using GLMakie

fig = Figure()

scatter(fig[1,1], rand(5), marker=Rect)
scatter(fig[1,2], rand(5), marker=Rect, strokewidth=5)
scatter(fig[2,1], rand(5), marker=:rect)
scatter(fig[2,2], rand(5), marker=:rect, strokewidth=5)
```

image

Note that the marker for `Rect` gets bigger as `strokewidth` increases. `strokecolor` has no effect in this case.

(`:circle` and `Circle` both work as expected)

### PR DESCRIPTION
**Pull request by ffreyer**
# Description

Fixes #2880, #473

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] Added an entry in NEWS.md (for new features and breaking changes)
- [ ] Added or changed relevant sections in the documentation
- [ ] Added unit tests for new algorithms, conversion methods, etc.
- [ ] Added reference image tests for new plotting functions, recipes, visual options, etc.

### CODE REVIEW

File: GLMakie/assets/shader/distance_shape.frag [hunk 1]
```diff
@@ -138,7 +138,7 @@ void main(){
     else if(shape == ROUNDED_RECTANGLE)
         signed_distance = rounded_rectangle(f_uv, vec2(0.2), vec2(0.8));
     else if(shape == RECTANGLE)
-        signed_distance = 1.0; // rectangle(f_uv);
```
Comment (SimonDanisch) [inline GLMakie/assets/shader/distance_shape.frag:141]: Hm, I remember I had all kind of artifacts with `rectangle(f_uv)` so I gave up and just used 1.0...
Will need to look at the reference images and try it out to see how bad it actually is.

File: GLMakie/assets/shader/sprites.geom [hunk 2]
```diff
@@ -164,11 +164,7 @@ void main(void)
     // Compute required amount of buffering
     float sprite_from_viewport_scale = 1.0 / viewport_from_sprite_scale;
     float bbox_buf = sprite_from_viewport_scale *
-                     (// Hack!! antialiasing is disabled for RECTANGLE==1 for now
-                      // because it's used for boxplots where the sprites are
-                      // long and skinny (violating assumption 1 above)
```
Comment (ffreyer) [inline GLMakie/assets/shader/sprites.geom:169]: I don't see a difference between master and this: (master | pr)
Screenshot from 2023-04-30 10-19-31Screenshot from 2023-04-30 10-14-30

Comment (ffreyer) [reply GLMakie/assets/shader/sprites.geom:169]: This is coming from https://github.com/JuliaPlots/GLMakie.jl/pull/30, but without images I don't know what this actually fixed.

**Comment by ffreyer** [comment]
That's a general issue I think. The default marker and `Circle` have this issue too: 
Screenshot from 2023-05-05 18-11-46
Probably a result of `f_viewport_from_u_scale` being a single value rather than an xy(z?) scale

**Pull request merged**